prepare next release

phryneas opened the pull request.

phryneas committed:
fix(tests): rework vue integration tests

- do not create a IncrementalChecker instance on the test thread
- use worker-rpc so the tests can actually access the checker instance run by the service
- fix include path in "tsconfig-imports.json" so the "should resolve *.vue in the same way as TypeScript" test actually compiles files
- add two counter checks to make sure the compiler actually throws errors it finds at all

phryneas committed:
feat(apiincrementalchecker): improve generation of diagnostics

when the options `checkSyntacticErrors: false` and `useTypescriptIncrementalApi: true` both were active, no semantic errors were emitted as soon as the first syntactic error was encountered
this patches the `typescript` import to override that behavior

see discussion in #257

phryneas committed:
fix(tests): linter tests - useTypescriptIncrementalApi usage

actually use the correct `useTypescriptIncrementalApi` value for some of the linter tests that were using the default value before

phryneas committed:
fix(tests): fix linter tests that were doing nothing

fix a bug where, as `tslint` was set to true and the files being checked had no parent folder with a tslint.json, tslint in these tests never found any warnings
also fixes a failing test because a syntactic error was counted as linter warning

semantic-release-bot committed:
chore(release): 1.2.0-beta.3 [skip ci]

# [1.2.0-beta.3](https://github.com/Realytics/fork-ts-checker-webpack-plugin/compare/v1.2.0-beta.2@beta...v1.2.0-beta.3@beta) (2019-04-22)

### Bug Fixes

* **tests:** rework vue integration tests ([5ad2568](https://github.com/Realytics/fork-ts-checker-webpack-plugin/commit/5ad2568))

phryneas committed:
Merge branch 'beta' into fix-linter-tests

phryneas committed:
Merge pull request #268 from phryneas/fix-linter-tests

improve generation of diagnostics, fix linter tests

semantic-release-bot committed:
chore(release): 1.2.0-beta.4 [skip ci]

# [1.2.0-beta.4](https://github.com/Realytics/fork-ts-checker-webpack-plugin/compare/v1.2.0-beta.3@beta...v1.2.0-beta.4@beta) (2019-04-23)

### Bug Fixes

* **tests:** fix linter tests that were doing nothing ([d078278](https://github.com/Realytics/fork-ts-checker-webpack-plugin/commit/d078278))
* **tests:** linter tests - useTypescriptIncrementalApi usage ([e0020d6](https://github.com/Realytics/fork-ts-checker-webpack-plugin/commit/e0020d6))

### Features

* **apiincrementalchecker:** improve generation of diagnostics ([ae80e5f](https://github.com/Realytics/fork-ts-checker-webpack-plugin/commit/ae80e5f)), closes [#257](https://github.com/Realytics/fork-ts-checker-webpack-plugin/issues/257)

swashata committed:
docs: 📖 add hooks documentation

Add documentation on how to use hooks on a compiler instance.

swashata committed:
docs: 📖 explain tapable hooks arguments

Explain the purpose of arguments passed to plugin hooks.

phryneas committed:
Merge pull request #274 from swashata/beta

docs: 📖 add hooks documentation

phryneas commented:
So we'd need to make a new release for the changes from #274 to land in the README. Also this lands a few test improvements and `feat(apiincrementalchecker): improve generation of diagnostics`.

Since the `beta` branch is protected I don't really know how to best prepare merging this into master @piotr-oles, can you help here?

piotr-oles commented:
I will take a look today’s evening :)

piotr-oles committed:
Merge branch 'master' into beta

piotr-oles commented:
I've resolved conflicts in the `CHANGELOG.md` - now everything should be good. I think semantic-release will release new beta version (1.2.1-beta.1).

semantic-release-bot committed:
chore(release): 1.3.0-beta.1 [skip ci]

# [1.3.0-beta.1](https://github.com/Realytics/fork-ts-checker-webpack-plugin/compare/v1.2.0...v1.3.0-beta.1@beta) (2019-04-30)

### Bug Fixes

* **tests:** fix linter tests that were doing nothing ([d078278](https://github.com/Realytics/fork-ts-checker-webpack-plugin/commit/d078278))
* **tests:** linter tests - useTypescriptIncrementalApi usage ([e0020d6](https://github.com/Realytics/fork-ts-checker-webpack-plugin/commit/e0020d6))
* **tests:** rework vue integration tests ([5ad2568](https://github.com/Realytics/fork-ts-checker-webpack-plugin/commit/5ad2568))

### Features

* **apiincrementalchecker:** improve generation of diagnostics ([ae80e5f](https://github.com/Realytics/fork-ts-checker-webpack-plugin/commit/ae80e5f)), closes [#257](https://github.com/Realytics/fork-ts-checker-webpack-plugin/issues/257)

phryneas commented:
Great! Who wants to pull the trigger? :smile:

johnnyreilly reviewed (dismissed):
First time with the semantic release hotness!

johnnyreilly commented:
It looks like we can't merge as Travis isn't running or something... @piotr-oles can you investigate please?

piotr-oles committed:
doc(changelog): update changelog spacing

piotr-oles commented:
We have all these problems because not all commits were merged into beta :) Travis didn't want to build because when we merged master to beta, semantic-release created new release and updated CHANGELOG.md with `[skip ci]` in the commit message. That why travis didn't want to start build. I have created a new commit with some spacing changes in the CHANGELOG.md to trigger build again without `[skip ci]` in the last commit :)

johnnyreilly commented:
Cool - is there a way we can avoid this happening in future? I'm keen we have a release process that just works™️ 😊

Is this happening just because this will be our first release end to end with semantic release? So in future these issues won't present? 

I'm a little confused / concerned with the problems we've been facing getting semantic release bedded in. Hopefully we can get past this and into a good release flow 🤞

swashata commented on README.md:
```diff
@@ -249,17 +249,110 @@ We hope this will be resolved in future; the issue can be tracked [here](https:/
+}).apply(compiler);
+// Now get the plugin hooks from compiler
+const tsCheckerHooks = ForkTsCheckerWebpackPlugin.getCompilerHooks(compiler);
+// These hooks provide acess to different events
```
Oops for the typo

```suggestion
// These hooks provide access to different events
```

swashata reviewed (changes requested):
I have found a typo in the docs. Please accept the changes in this PR too :)

piotr-oles commented:
The most problematic is generation of the `CHANGELOG.md` - I think that's why it's not in the default semantic-release setup. After we merge to the `master` it creates additional commit with updated `CHANGELOG.md`. This commit doesn't appear in the `beta` branch and this creates merge conflicts in the next pull request (as we modify `CHANGELOG.md` in every beta release). We can manualy merge the master barnch into the beta after release but maybe we could automate this also (that was the point of the semantic-release)

swashata committed:
doc: fix typo in the hooks documentation

Co-Authored-By: piotr-oles <piotrek.oles@gmail.com>

johnnyreilly commented:
My take away from that is that semantic release doesn't have a good workflow with regards to changelogs. That's super disappointing. 

I'm rather hesitant about us creating our own automation around this as that may end up being another thing to maintain as semantic release evolves. If there's not a good out of the box experience that we can have I'm inclined to say "well nevermind - let's just have the GitHub releases instead"

It means we have to be online to see them and if we were ever to move the plugin elsewhere we'd need to recreate manually. Neither of those is a deal-breaker in my eyes.

I value a simple release workflow above other concerns. If you want to drop the changelog generation I'm fine with that. 😊

phryneas commented:
Since the CHANGELOG is automatically generated from commit messages, I think we won't lose any metadata when dropping the automatic generation - even if this project were to move away from github some day in the future. The commit history won't be lost.

Commit:
chore: remove changelog.md update from semantic-release

Commit:
chore: remove emoji from commits to improve consistency

Not everyone uses emoji in commits - so it's better to remove them than
having two styles of commits

piotr-oles commented:
I've created PR to drop `CHANGELOG.md` generation and remove emoji as not everyone is using them 😢

piotr-oles commented:
The last thing we should remember is to use "Rebase and merge" when merging from the `beta` to `master` - that will not create additional commit and will create a tag with the new version on the last commit that was also on the beta branch. Thanks to that master and beta will be in sync (otherwise master will be 1 commit ahead of beta - the merge commit). I will create PR with docs update for the deployment process.

johnnyreilly commented:
Good catch - thanks!

Commit:
docs: update release documentation

piotr-oles commented:
:tada: This PR is included in version 1.3.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/fork-ts-checker-webpack-plugin)
- [GitHub release](https://github.com/Realytics/fork-ts-checker-webpack-plugin/releases/tag/v1.3.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

johnnyreilly commented:
How did you get it to work? Can you just not rebase on a phone?

phryneas commented:
Nope, looks like this was one last normal merge. I guess we'll have to rebase that one back to beta one last time and then we'll be golden as long as we don't push to master.
